ptvsd/_vendored/pydevd/_pydevd_bundle/pydevd_comm.py

```diff
@@ -587,19 +571,86 @@ class NetCommand:
     or one to be sent by daemon.
     """
     next_seq = 0 # sequence numbers
-
-    def __init__(self, id, seq, text):
-        """ smart handling of parameters
-        if sequence is 0, new sequence will be generated
-        if text has carriage returns they'll be replaced"""
-        self.id = id
+    
+    # Protocol where each line is a new message (text is quoted to prevent new lines).
+    QUOTED_LINE_PROTOCOL = 'quoted-line'
+    
+    # Uses http protocol to provide a new message.
+    # i.e.: Content-Length:xxx\r\n\r\npayload
+    HTTP_PROTOCOL = 'http'
+    
+    protocol = QUOTED_LINE_PROTOCOL
+    
+    _showing_debug_info = 0
+    _show_debug_info_lock = threading.RLock()
+
+    def __init__(self, cmd_id, seq, text):
+        """
+        If sequence is 0, new sequence will be generated (otherwise, this was the response
+        to a command from the client).
+        """
+        self.id = cmd_id
         if seq == 0:
             NetCommand.next_seq += 2
             seq = NetCommand.next_seq
         self.seq = seq
-        self.text = text
-        encoded = quote(to_string(text), '/<>_=" \t')
-        self.outgoing = '%s\t%s\t%s\n' % (id, seq, encoded)
+        
+        if IS_PY2:
+            if isinstance(text, unicode):
+                text = text.encode('utf-8')
+        else:
+            assert isinstance(text, str)
```

Comment:
Looks like this assert should be outside of `if`?

Reply:
Or rather there should be some assert in the above branch that does the equivalent check there (in case `if` doesn't get triggered).

Reply:
Done.